marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs

```diff
@@ -274,24 +289,26 @@ epochStateWorker_
         nodeConfigPath
         onInsert
         securityParam
-        Coordinator{_barrier}
+        Coordinator{_barrier, _errorVar}
         ch
         dbPath = do
     nodeConfigE <- runExceptT $ readNetworkConfig (NetworkConfigFile nodeConfigPath)
-    nodeConfig <- either (error . show) pure nodeConfigE
+    nodeConfig <- either (throw . CantStartIndexer . Text.pack . show) pure nodeConfigE
```

Comment:
So this throws a runtime error, right? It's not caught anywhere?

If it's thrown, the thread stops, but the application still continues to run right?

Reply:
This one aren't using ExceptT yet, not sure if we want to introduce it here as well as we didn't start the indexers yet (we're about to start them).
This code is before any `forkIO` call, so it leaves in the "main" thread, and an uncaught exception will cause a failure of the main thread and thus of the whole application.